src/main/java/org/blyznytsia/validator/DependencyValidator.java

```diff
@@ -0,0 +1,93 @@
+package org.blyznytsia.validator;
+
+import org.blyznytsia.exception.CircularDependencyException;
+import org.blyznytsia.model.BeanDefinition;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.LinkedList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+
+/**
+ * The main purpose of this class is to find the circular dependency between beans.
+ * When problem occurs than message with information about chain of beans is printed.
+ * If circular dependency doesn't exist then return empty set.
+ */
+public class DependencyValidator {
+
+    /**
+     * Map of beans. Key is a bean, and value is a list of beans in which the key bean depends.
+     */
+    private final Map<String, List<String>> beans;
+
+    private DependencyValidator(Map<String, List<String>> beans) {
+        this.beans = beans;
+    }
+
+    public static DependencyValidator buildDependencyGraph(List<BeanDefinition> beanDefinitions) {
+
+        Map<String, List<String>> beanDeps = beanDefinitions
+            .stream()
+            .collect(Collectors.toMap(
+                BeanDefinition::getName,
+                BeanDefinition::getDependsOnBeans)
+            );
+
+        return new DependencyValidator(beanDeps);
+    }
+
+    /**
+     * Calculate cycles between beans.
+     *
+     * @return empty set or {@link CircularDependencyException}
+     */
+    public Set<String> calculateCycles() {
```

Comment:
The interface exposed by the validator should be `validate`. 
No need to return an empty set ... it should take a list of bean definitions and throw an exception if there are any violations (e.g. cyclic deps)

Reply:
Absolutely agree!